### PR TITLE
Feat/redesign modals

### PR DESCRIPTION
**Pull request by seibei-iguchi**
Updated several modals including the Ethereum staking flow with the new modal component. Refactor the content to use components from the design system. Updated the Paragraph component with variants and frame props.

### Before:
Screenshot 2024-08-28 at 17 39 40
Screenshot 2024-08-28 at 17 36 54
Screenshot 2024-08-28 at 17 37 11
Screenshot 2024-08-28 at 17 37 23
Screenshot 2024-08-28 at 17 37 33

### After:
Screenshot 2024-08-26 at 10 22 12
Screenshot 2024-08-28 at 15 45 12
Screenshot 2024-08-28 at 16 59 24
Screenshot 2024-08-29 at 9 41 10
Screenshot 2024-08-28 at 16 59 55
Screenshot 2024-08-28 at 17 22 47

### CODE REVIEW

File: packages/components/src/components/NewModal/NewModalBackdrop.tsx [hunk 1]
```diff
@@ -27,6 +27,10 @@ const Wrapper = styled.div<{ $alignment: NewModalAlignment; $padding: number }>`
     background: rgb(0 0 0 / 30%);
     align-items: ${({ $alignment }) => mapAlignmentToAlignItems($alignment)};
     justify-content: ${({ $alignment }) => mapAlignmentToJustifyContent($alignment)};
+
+    > * + * {
```
Comment (jvaclavik) [inline packages/components/src/components/NewModal/NewModalBackdrop.tsx:31]: Why do you need this? Looks like a magic 😅

File: packages/components/src/components/typography/Paragraph/Paragraph.stories.tsx [hunk 2]
```diff
@@ -25,5 +27,11 @@ export const Paragraph: StoryObj<ParagraphProps> = {
                 'label',
             ],
         },
+        variant: {
+            control: {
+                type: 'select',
+            },
+            options: ['primary', 'secondary', 'tertiary', 'info', 'warning', 'destructive'],
```
Comment (jvaclavik) [inline packages/components/src/components/typography/Paragraph/Paragraph.stories.tsx]: Could you use it this way? 

```tsx
// component: 
export const paragraphVariants = [
    'primary', 'secondary', 'tertiary', 'info', 'warning', 'destructive'
] as const;

export type ParagraphVariant = Extract<UIVariant, (typeof paragraphVariants)[number]>;

// storybook 

option: paragraphVariants
```

Check other component variants in storybooks

File: packages/components/src/components/typography/Paragraph/Paragraph.stories.tsx [hunk 3]
```diff
@@ -25,5 +27,11 @@ export const Paragraph: StoryObj<ParagraphProps> = {
...
+        },
```
Comment (jvaclavik) [inline packages/components/src/components/typography/Paragraph/Paragraph.stories.tsx]: please add:

```tsx
args: {
  ...,
  ...getFramePropsStory(allowedParagraphFrameProps).args,
},
argTypes: {
  ...,
  ...getFramePropsStory(allowedParagraphFrameProps).argTypes,
}

File: packages/components/src/components/NewModal/NewModalBackdrop.tsx (hunk 1 above)
Comment (seibei-iguchi) [reply packages/components/src/components/NewModal/NewModalBackdrop.tsx:31]: I couldn't help myself:D It's the deranged owl selector. I'll just use gap:))

Comment (jvaclavik) [reply packages/components/src/components/NewModal/NewModalBackdrop.tsx:31]: Yes please :D

File: packages/components/src/components/Flex/Flex.tsx [hunk 4]
```diff
@@ -40,7 +40,14 @@ export const flexAlignItems = [
...
 export type FlexAlignItems = (typeof flexAlignItems)[number];
-export type Flex = string | number;
+export type Flex =
+    | 'none'
+    | 'auto'
+    | 'initial'
+    | 'inherit'
+    | `${number}`
+    | `${number} ${number}`
+    | `${number} ${number} ${string}`;
```
Comment (peter-sanderson) [inline packages/components/src/components/Flex/Flex.tsx:51]: :clap:

File: packages/suite/src/support/messages.ts [hunk 5]
```diff
@@ -8739,7 +8739,7 @@ export default defineMessages({
     },
     TR_STAKE_YOUR_FUNDS_MAINTAINED: {
         id: 'TR_STAKE_YOUR_FUNDS_MAINTAINED',
-        defaultMessage: 'Your staked funds are <h>maintained by Everstake</h>',
```
Comment (peter-sanderson) [inline packages/suite/src/support/messages.ts:8742]: this `<h>`, etc.. is supposed to be removed?

Comment (seibei-iguchi) [reply packages/suite/src/support/messages.ts:8742]: Yep, I have to do that in Crowdin

File: packages/components/src/components/typography/Paragraph/Paragraph.tsx [hunk 6]
```diff
@@ -1,26 +1,76 @@
...
 
-const P = styled.div<{ $typographyStyle: TypographyStyle }>`
+const mapVariantToColor = ({
+    $variant,
+    theme,
+}: {
+    $variant: ParagraphVariant;
+    theme: DefaultTheme;
+}): CSSColor => {
+    const colorMap: Record<ParagraphVariant, Color> = {
```
Comment (peter-sanderson) [inline packages/components/src/components/typography/Paragraph/Paragraph.tsx]: nit: map can be outside of the function

**Review by peter-sanderson — APPROVED** [review]

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**